tests/unit/Models.Vulnerability.spec.js

```diff
@@ -0,0 +1,140 @@
+'use strict'
+
+/*!
+This file is part of CycloneDX JavaScript Library.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+Copyright (c) OWASP Foundation. All Rights Reserved.
+*/
+
+const assert = require('assert')
+const { suite, test } = require('mocha')
+const { PackageURL } = require('packageurl-js')
+
+const {
+  Enums: {
+    VulnerabilityAnalysisState
+  },
+  Models: {
+    OrganizationalEntity,
+    OrganizationalEntityRepository,
+    Property,
+    PropertyRepository,
+    Source,
+    Tool,
+    ToolRepository,
+    Vulnerability,
+    VulnerabilityAdvisory,
+    VulnerabilityAdvisoryRepository,
+    VulnerabilityAffects,
+    VulnerabilityAffectsRepository,
+    VulnerabilityAnalysis,
+    VulnerabilityAnalysisRepository,
+    VulnerabilityCredits,
+    VulnerabilityRating,
+    VulnerabilityRatingRepository,
+    VulnerabilityReference,
+    VulnerabilityReferenceRepository
+  }
+} = require('../../')
+
+suite('Models.Vulnerability', () => {
+  test('constructor', () => {
+    const vulnerability = new Vulnerability()
+
+    assert.strictEqual(vulnerability.bomRef.value, undefined)
+    assert.strictEqual(vulnerability.id, undefined)
+    assert.strictEqual(vulnerability.source, undefined)
+    assert.strictEqual(vulnerability.references.size, 0)
```

Comment:
:x: please add type checks -- as a regression test to find unintended type chages (which would lead to breaking changes)